src/three-components/TextureUtils.js

```diff
@@ -45,21 +50,80 @@ export default class TextureManager extends EventDispatcher {
     super();
     this.config = {...defaultConfig, ...config};
     this.renderer = renderer;
-    this.cubemapGenerator = new EquirectangularToCubemap(this.renderer);
     this.envMapGenerator = new EnvMapGenerator(this.renderer);
   }
 
+  /**
+   * The texture returned here is from a WebGLRenderCubeTarget,
+   * which is not the same as a THREE.CubeTexture, and just what
+   * the current THREE.CubeCamera uses, and has the same effect
+   * when being used as an environment map.
+   *
+   * @param {THREE.Texture} texture
+   * @return {THREE.Texture}
+   */
+  equirectangularToCubemap(texture) {
+    // Each instance of EquirectangularToCubeGenerator creates a new camera,
+    // scene and render target. Every time a scene/camera duo is rendered,
+    // a new three.js WebGLRenderState is cached in the renderer, with no way
+    // of draining, causing a memory leak. The render target isn't shareable
+    // between textures either, as the caller of this function is responsible
+    // for disposing the texture. To get around these constraints, create
+    // new generators, but reuse the same camera and scene as to not generate
+    // extra WebGLRenderStates in cache.
+    const generator = new EquirectangularToCubeGenerator(texture, {
+      resolution: this.config.cubemapSize,
+    });
+
+    // If this is the first generator, stub out the getShader method
+    // so we don't create unnecessary shaders; we're continuing to
+    // reuse the same camera and scene after all. Necessary to overwrite
+    // the prototype since this is called in the constructor.
+    if (!this[$cubeGenerator]) {
+      const material = generator.boxMesh.material;
+      EquirectangularToCubeGenerator.prototype.getShader = () => material;
```

Comment:
Is it possible to just create an extended class rather than monkey patching on the fly here? e.g.,

```javascript
class NonLeakyEquirectangularToCubeGenerator extends EquirectangularToCubeGenerator {
  constructor(sharedShaderMaterial = null) {
    super();
    this.sharedShaderMaterial = sharedShaderMaterial;
  }

  getShader() {
    if (this.sharedShaderMaterial != null) {
      return this.sharedShaderMaterial;
    }
    return super.getShader();
  }
}
```

Reply:
Good call, trying this